issues.py

```diff
@@ -350,19 +334,15 @@ def sync_repo(repo_name):
     cs_alerts = {repo_name + '/code_scanning/' + str(a['number']): a for a in get_alerts(repo_name)}
 
     # fetch issues from JIRA and delete duplicates and ones which can't be matched
-    issue_search = 'project={jira_project} and description ~ "\\"GH_ALERT_LOOKUP={repo_name}/\\""'.format(
-        jira_project=JIRA_PROJECT,
-        repo_name=repo_name,
-    )
     jira_issues = {}
-    for i in jira.search_issues(issue_search, maxResults=0):
+    for i in fetch_issues(repo_name):
         key = get_issue_id(i)
         if key in jira_issues:
             app.logger.info('Deleting duplicate jira alert issue.')
-            i.delete()
+            i.delete()   # TODO - seems scary, are we sure....
```

Comment:
I guess it could be dangerous if a lookup string (`GH_ALERT_LOOKUP=...`) somehow ends up in a non-alert issue's body, which can easily happen when someone copies the body of an alert issue.
Does the customer that you are working with have a separate JIRA project for alert issues or would they just dump those into existing projects where they would be mixed with human-created issues?

I guess there are various other ways in which we could be more careful with duplicate and orphaned issues. But they will either slightly confuse the user or will be more difficult to implement. The current implementation seemed straightforward, that's why I chose it, but I agree with the concerns.

Reply:
We could also add a magic prefix into each alert-issue's title, such as `[CODE SCANNING ALERT]`. Surely, user-managed issues will not have such a prefix. We could in addition prominently document which issues will be managed by our application.

If someone then manually creates an issue with such a title prefix AND manages to copy the lookup string into the body, then we can just tell them `"told you so"`.